blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -230,7 +227,7 @@ public HttpResponse validateAndCreate(@JsonBody JSONObject request) {
             //Now we know the token is valid. Lets find credential
             StandardUsernamePasswordCredentials githubCredential = CredentialsUtils.findCredential(getId(), StandardUsernamePasswordCredentials.class, new BlueOceanDomainRequirement());
 
-            final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, "github", "Github Access Token", authenticatedUser.getId(), accessToken);
+            final StandardUsernamePasswordCredentials credential = new UsernamePasswordCredentialsImpl(CredentialsScope.USER, getId(), "Github Access Token", authenticatedUser.getId(), accessToken);
```

Comment:
I think we need to override `getId` in `GithubEnterpriseScm` to return the `apiUrl`. This would allow you to support multiple Github Enterprise servers that have entirely different credentials.